function.k

```diff
@@ -0,0 +1,231 @@
+	        <inputParams> Ps </inputParams>
+	        <Returns> Rs </Returns>
+	        <quantifiers> FQ </quantifiers>
+	        <visibility> .K </visibility>
+	        <body> .Statements </body>
+            </Function>)
+         ...
+    </Contract>
+
+rule
+    <k> #pcsContractPart(Ct:Id, function F:Id (Ps:FParameters) FQ:FunQuantifiers;)
+            => . ... </k>
+    <Contract>
+        <cname> Ct </cname>
+        (.Bag =>
+            <Function>
+	        <fname> F </fname>
+	        <inputParams> Ps </inputParams>
+	        <Returns> .FParameters </Returns>
+	        <quantifiers> FQ </quantifiers>
+	        <visibility> .K </visibility>
+	        <body> .Statements </body>
+            </Function>)
+         ...
+    </Contract>
+
+rule 
+    <k> #pcsContractPart(Ct:Id, constructor (Ps:FParameters) FQ:FunQuantifiers {B}) => . ... </k>
+    <Contract>
+        <cname> Ct </cname>
+        (.Bag =>
+            <Function>
+	        <fname> String2Id("constructor") </fname>
+	        <inputParams> Ps </inputParams>
+	        <Returns> .FParameters </Returns>
+	        <quantifiers> FQ   </quantifiers>
+	        <visibility>  .K </visibility>
+	        <body> B </body>
+            </Function>)
+         <Constructor> _ => true </Constructor>
+         ...
+    </Contract>
+
+rule
+    <k> #pcsContractPart(Ct:Id, function (Ps:FParameters) FQ:FunQuantifiers {B})
+            => . ... </k>
+    <Contract>
+         <cname> Ct </cname>
+         (.Bag =>
+             <Function>
+	         <fname> String2Id("fallback") </fname>
+	         <inputParams> Ps </inputParams>
+	         <Returns> .FParameters </Returns>
+	         <quantifiers> FQ </quantifiers>
+	         <visibility> .K </visibility>
+	         <body> B </body>
+             </Function>)
+          ...
+    </Contract>
+
+
+syntax SolidityItem 
+       ::= #CallBody(Statements)  
+         | #Call(Id,Values)
+         | #InitParams(FParameters, Values)
+         | #InitParam(FParameter, Value)
+         | "#return"
+         | "#returnContract"
+         | #BindInputParams(FParameters, Values)
+         | #BindReturnParams(FParameters, Values)
+
+rule [External-Function-Call]:
+     <k> functionCall(C:Int; F:Id; Vs:Values) => #Call(F,Vs)  ~> #returnContract ...</k>
+     <currentAccount> #account(C1,N1) => #account(C,N) </currentAccount>
+     <env> Rho:Map => CRho </env>
+     <callStack> .List => ListItem(#state(#account(C1,N1), Rho, #return(false,#undef_Value), #external)) ... </callStack>
+     <callDepth> Dep => Dep +Int 1 </callDepth>
+     <account>
+         <acctID> #account(C,N) </acctID>
+         <acctEnv> CRho:Map </acctEnv>
+         ...
+     </account>
+
+
+rule [Internal-Function-Call]:
+     <k> functionCall(F:Id; Vs:Values) => #Call(F,Vs) ~> #return ... </k>
+     <currentAccount> #account(C,N) </currentAccount>
+     <env> ENV:Map  </env>
+     <callStack> .List => ListItem(#state(#account(C,N),ENV,#return(false,#undef_Value), #internal)) ... </callStack> 
```

Comment:
I think the `<callStack>` should be modeled more directly off of the KEVM one, so that when you push to the `<callStack>` all you have to do is put a copy of the entire `<callState>` sub-configuration onto the front of the stack. It makes these rules very simple, and makes it so that it's easy to decide whether a part of the configuration belongs in the `<callState>` sub-configuration: If each call-frame (ie. elements of the `<callStack>`) needs its own copy of a piece of state then add that piece of state inside the `<callState>` sub-configuration, if not then put it outside that sub-configuration.